annalieseschropp/checkers
# 2 Name Entry

What was Accomplished

- The name entry scene of the project was created in the NameEntry scene.
- The name entry scene includes:
    - 2 drop down menus, for previous names stored in records
    - 2 input fields, to get new player names
    - 1 toggle button, for if the game will be played with forced capture or not
    - 2 buttons, Play Game and Cancel buttons
        - Play Game button stores all the data in the static class to pass between scenes
        - Cancel button takes the player back to the main menu
- Unit Testing, passes all 7 tests
     - Testing all possible paths through the program, checking if the error handling is proper and if accepted inputs are stored correctly in the static class.

Testing

To run unit tests, ensure you have a build set for the NameEntry scene
- File > Build Settings > Add Open Scenes > Build
You also need the Test Runner window open to see the available tests
- Window > General > Test Runner
- You can dock this where you would like based on your layout preferences
Click Run All in the Test Runner window or select a specific test then click Run Selected
To manually run the scene, press the play button or go to File > Build and Run

To Dos:

- Connect this scene to the main menu and the game board.
- Add options to drop down menu to select previous names using the records.
- Connect the static script to game board to pass information to the game board scene
